[7_2_X][TIMOB-26003] Android: Provide API to support opt-out of Analytics at runtime

ypbnv opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26003

**Description:**
Backport of:https://github.com/appcelerator/titanium_mobile/pull/10058

ypbnv committed:
Expose optOut property for Analytics.

ypbnv committed:
Fix formatting.

build commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10059/1/artifact/dist/mobilesdk-7.2.0.v20180517072335-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

ypbnv committed:
Update with the latest aps_analytics jar.

ypbnv committed:
Move appEnrollEvent sending after evaluation of app.js
Remove sending of geoEvent.
Ensure Analytics are initialized before app.js evaluation, but before the first event is sent.

ypbnv committed:
Add latest version of aps-analytics.

Commit:
[TIMOB-26003] Remove redundant Geolocation doAnalytics()

Commit:
[TIMOB-26027] Update aps-analytics.jar

Commit:
[TIMOB-26027] Update AndroidManifest.xml

build commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :thumbsup: Hey!, You deleted more code than you added. That's awesome!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

Commit:
Update with latest changes.

ypbnv commented:
Updated with the latest changes in master PR.

garymathews reviewed (approved):
CR: **PASS**
